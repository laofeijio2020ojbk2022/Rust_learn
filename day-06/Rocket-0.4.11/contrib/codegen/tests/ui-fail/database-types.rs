extern crate rocket;
#[macro_use] extern crate rocket_contrib;

struct Unknown;

#[database("foo")]
struct A(Unknown);
//~^ ERROR Unknown: rocket_contrib::databases::Poolable

#[database("foo")]
struct B(Vec<i32>);
//~^ ERROR Vec<i32>: rocket_contrib::databases::Poolable

fn main() {  }
