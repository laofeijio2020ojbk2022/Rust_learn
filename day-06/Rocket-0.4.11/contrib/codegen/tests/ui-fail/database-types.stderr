error[E0277]: the trait bound `Unknown: Poolable` is not satisfied
 --> $DIR/database-types.rs:7:10
  |
7 | struct A(Unknown);
  |          ^^^^^^^ the trait `Poolable` is not implemented for `Unknown`

error[E0277]: the trait bound `Vec<i32>: Poolable` is not satisfied
  --> $DIR/database-types.rs:11:10
   |
11 | struct B(Vec<i32>);
   |          ^^^^^^^^ the trait `Poolable` is not implemented for `Vec<i32>`
